### DISCUSSION TITLE
Why is 2 windows opened while using Driver with options?

### DISCUSSION THREAD

**Discussion by coskundeniz** [seleniumbase/SeleniumBase#2373] [Q&A] [answered]
This is the example code I use:

```
from seleniumbase import Driver
from seleniumbase.undetected.options import ChromeOptions

chrome_options = ChromeOptions()
chrome_options.add_argument("--disable-gpu")
chrome_options.add_argument("--disable-dev-shm-usage")
chrome_options.add_argument("--disable-infobars")
chrome_options.add_argument("--disable-popup-blocking")
chrome_options.add_argument("--disable-notifications")
chrome_options.add_argument("--ignore-ssl-errors")
chrome_options.add_argument("--no-first-run")
chrome_options.add_argument("--no-service-autorun")
chrome_options.add_argument("--disable-blink-features=AutomationControlled")
chrome_options.add_argument("--disk-cache-size=0")
chrome_options.add_argument(f"--user-agent={user_agent}")

driver = Driver(uc=True, headed=True)
driver.options = chrome_options

driver.get("https://nowsecure.nl")
print(f"Title: {driver.get_title()}")
sleep(5)
driver.quit()
```

![Screenshot from 2023-12-19 13-32-46](https://github.com/seleniumbase/SeleniumBase/assets/4516210/c8357ae8-af1e-4873-9960-6f0582f3bbb3)

Thank you.

**Comment by mdmintz** [comment, marked as answer]
`seleniumbase.undetected` is not designed to be called directly. (See valid formats: [SeleniumBase/help_docs/syntax_formats.md](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md))

### For the direct `Driver()` format, pass your options via the built-in method args.

https://github.com/seleniumbase/SeleniumBase/blob/570910c6f6e079cddd00c23e826e202c6489f39e/seleniumbase/plugins/driver_manager.py#L65

### Most of the options you're trying to pass are already automatically being set as needed:

https://github.com/seleniumbase/SeleniumBase/blob/570910c6f6e079cddd00c23e826e202c6489f39e/seleniumbase/core/browser_launcher.py#L1033

https://github.com/seleniumbase/SeleniumBase/blob/570910c6f6e079cddd00c23e826e202c6489f39e/seleniumbase/core/browser_launcher.py#L1035

https://github.com/seleniumbase/SeleniumBase/blob/570910c6f6e079cddd00c23e826e202c6489f39e/seleniumbase/core/browser_launcher.py#L1112

https://github.com/seleniumbase/SeleniumBase/blob/570910c6f6e079cddd00c23e826e202c6489f39e/seleniumbase/core/browser_launcher.py#L1121

https://github.com/seleniumbase/SeleniumBase/blob/570910c6f6e079cddd00c23e826e202c6489f39e/seleniumbase/core/browser_launcher.py#L2460

https://github.com/seleniumbase/SeleniumBase/blob/570910c6f6e079cddd00c23e826e202c6489f39e/seleniumbase/undetected/__init__.py#L234

https://github.com/seleniumbase/SeleniumBase/blob/570910c6f6e079cddd00c23e826e202c6489f39e/seleniumbase/undetected/__init__.py#L235

Correct usage should resolve any problems you experience. Note that when opening a URL that has bot-detection software on it, SeleniumBase opens a clean tab/window (and closes the existing one).

**Comment by coskundeniz** [comment]
Thank you for the answer.
